### PR TITLE
add saturn integration

### PR DESCRIPTION
**Pull request by salieflewis**
TL;DR of the Saturn stuff is we can speed up all of our content addressed media (except for video and audio, which we're leaving for Mux). They're in private beta we won't be charged for usage.

The way it works is by using a service worker, which is what this PR configures in our root layout, then sets up a reverse proxy for our domain. Essentially the service worker looks for requests with cids in them, and routes them to Saturn instead. If the request to Saturn fails, it just falls back to whatever link/gateway we've got set up, so no harm there.

I'm eager to see how effectively this speeds things up!

### CODE REVIEW

**Review by 0xTranqui — APPROVED** [review]
idk how this works but send it!

**Review by jawndiego — APPROVED** [review]

**Comment by salieflewis** [comment]
> idk how this works but send it!

Lets see if it starts working in prod. No harm either way, we can always remove it.
